models/magiceden/solana/magiceden_solana_trades.sql

```diff
@@ -0,0 +1,243 @@
+                    then rpad(bytearray_substring(call_data,1+16,2), 8, 0xff) else bytearray_substring(call_data,1+16,2) end)) as makerFeeRaw
+                , cast(json_value(args, 'strict $.MIP1ExecuteSaleV2Args.takerFeeBp') as double) as takerFeeBp
+                , cast(bytearray_to_bigint(bytearray_reverse(bytearray_substring(call_data,1+18,2))) as double) as takerFeeRaw
+                , call_outer_instruction_index
+                , call_inner_instruction_index
+                , call_block_time
+                , call_block_slot
+                , call_tx_id
+                , call_tx_signer
+                , row_number() over (partition by call_tx_id order by call_outer_instruction_index asc, call_inner_instruction_index asc) as call_order
+            FROM {{ source('magic_eden_solana','m2_call_mip1ExecuteSaleV2') }}
+            UNION ALL
+            SELECT 
+                call_instruction_name
+                , account_buyer
+                , account_seller
+                , account_metadata
+                , account_tokenMint
+                --again, some transactions are missing "args"
+                , coalesce(cast(json_value(args, 'strict $.OCPExecuteSaleV2Args.price') as double)
+                    , cast(bytearray_to_bigint(bytearray_reverse(bytearray_substring(call_data,1+8,8))) as double)
+                    ) as buyerPrice
+                , 1 as tokenSize
+                , cast(json_value(args, 'strict $.OCPExecuteSaleV2Args.makerFeeBp') as double) as makerFeeBp
+                , bytearray_to_bigint(bytearray_reverse(case when bytearray_substring(call_data,1+17,1) = 0xff --check second byte for 0xff
+                    then rpad(bytearray_substring(call_data,1+16,2), 8, 0xff) else bytearray_substring(call_data,1+16,2) end)) as makerFeeRaw
+                , cast(json_value(args, 'strict $.OCPExecuteSaleV2Args.takerFeeBp') as double) as takerFeeBp
+                , cast(bytearray_to_bigint(bytearray_reverse(bytearray_substring(call_data,1+18,2))) as double) as takerFeeRaw
+                , call_outer_instruction_index
+                , call_inner_instruction_index
+                , call_block_time
+                , call_block_slot
+                , call_tx_id
+                , call_tx_signer
+                , row_number() over (partition by call_tx_id order by call_outer_instruction_index asc, call_inner_instruction_index asc) as call_order
+            FROM {{ source('magic_eden_solana','m2_call_ocpExecuteSaleV2') }}
+        ) trade
+        --this shortcut ONLY works if you know that a log is only emitted ONCE per call.
+        LEFT JOIN royalty_logs rl ON trade.call_tx_id = rl.call_tx_id
+            AND trade.call_block_slot = rl.call_block_slot
+            AND trade.call_order = rl.log_order 
+        {% if is_incremental() %}
+        WHERE trade.call_block_time >= NOW() - interval '7' day
+        {% endif %}
```

Comment:
let's move this into the trade CTE on each read instead. similar to above comments, lets use the new incremental macro